kernel/syscall.c

```diff
@@ -2815,20 +2815,32 @@ long myst_syscall_prlimit64(
     if (pid)
         return -EINVAL;
 
-    // Only support resource NOFILE
-    if (resource != RLIMIT_NOFILE)
-        return -EINVAL;
+    if (resource == RLIMIT_NOFILE)
+    {
+        if (old_rlim)
+        {
+            // ATTN: return currently effective RLIMIT_NOFILE settings
+            old_rlim->rlim_cur = 65536;
+            old_rlim->rlim_max = 65536;
+        }
 
-    if (old_rlim)
+        if (new_rlim)
+        {
+            // ATTN: make RLIMIT_NOFILE settings effective ;
+        }
+    }
+    else if (resource == RLIMIT_STACK)
     {
-        // ATTN: return currently effective RLIMIT_NOFILE settings
-        old_rlim->rlim_cur = 65536;
-        old_rlim->rlim_max = 65536;
+        if (old_rlim)
+        {
+            // TODO: get actual stack limits
+            old_rlim->rlim_cur = 1024 * 1024 * 1024;
+            old_rlim->rlim_max = 1024 * 1024 * 1024;
```

Comment:
```suggestion
            old_rlim->rlim_max = MYST_PROCESS_INIT_STACK_SIZE;
```
MYST_PROCESS_INIT_STACK_SIZE is defined in the main branch, please rebase.